core/templates/tests/unit-test-utils.ts

```diff
@@ -0,0 +1,139 @@
+import { NgZone, PlatformRef, Type } from '@angular/core';
+import { platformBrowserDynamic } from '@angular/platform-browser-dynamic';
+import { BrowserModule } from '@angular/platform-browser';
+import { downgradeComponent, UpgradeModule } from '@angular/upgrade/static';
+
+declare var angular: ng.IAngularStatic;
+/**
+ * Returns a div containing the compiled html.
+ * @param {string} html - The html to be compiled.
+ */
+export const html = (html: string): Element => {
+  // Don't return `body` itself, because using it as a `$rootElement` for ng1
+  // will attach `$injector` to it and that will affect subsequent tests.
+  const body = document.body;
+  body.innerHTML = `<div>${html.trim()}</div>`;
+  const div = document.body.firstChild as Element;
+
+  if (div.childNodes.length === 1 && div.firstChild instanceof HTMLElement) {
+    return div.firstChild;
+  }
+
+  return div;
+};
+
+/**
+ * Trims the text by removing new lines and extra white space.
+ * @param {string | null | undefined} text - The text to be trimmed of spaces.
+ * @param {bool} allSpace - A bool to define whether to replace new line
+ *   with a space or empty string.
+ */
+export const multiTrim = (
+    text: string | null | undefined, allSpace = false): string => {
+  // eslint-disable-next-line eqeqeq
+  if (typeof text == 'string') {
+    const repl = allSpace ? '' : ' ';
+    return text.replace(/\n/g, '').replace(/\s+/g, repl).trim();
+  }
+  throw new Error('Argument can not be undefined.');
+};
+
+/**
+ * And upgraded angularjs module.
+ * @param {PlatformRef} platform - The platform reference for bootstraping.
+ * @param {Type<{}>} Ng2Module - Angular module to bootstrap.
+ * @param {Element} element - An html element.
+ * @param {angular.IModule} ng1Module - The angularjs module to be included
+ *  in angular zone.
+ */
+export const bootstrap = (
+    platform: PlatformRef,
+    Ng2Module: Type<{}>,
+    element: Element,
+    ng1Module: angular.IModule
+): Promise<UpgradeModule> => {
+// We bootstrap the Angular module first; then when it is ready (async) we
+// bootstrap the AngularJS module on the bootstrap element (also ensuring that
+// AngularJS errors will fail the test).
+  return platform.bootstrapModule(Ng2Module).then(ref => {
+    const ngZone = ref.injector.get<NgZone>(NgZone);
+    const upgrade = ref.injector.get(UpgradeModule);
+    const failHardModule = ($provide) => {
+      $provide.value('$exceptionHandler', (err) => {
+        throw err;
+      });
+      return '';
+    };
+
+    // The `bootstrap()` helper is used for convenience in tests, so that we
+    // don't have to inject and call `upgrade.bootstrap()` on every Angular
+    // module. In order to closer emulate what happens in real application,
+    // ensure AngularJS is bootstrapped inside the Angular zone.
+    ngZone.run(() => upgrade.bootstrap(
+      // This throws "Type '($provide: any) => string' is not assignable to
+      // type 'string'."
```

Comment:
Also, can you add a comment explaining why we need `ts-ignore` here.

Reply:
Done!